src/main/java/processor/Processor.java

```diff
@@ -0,0 +1,268 @@
+package processor;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import storage.Storage;
+import tasks.Task;
+import tasks.TaskList;
+import ui.Ui;
+
+/**
+ * The Processor class is responsible for processing user input and performing operations on the TaskList.
+ * It interacts with the TaskList, Ui, and Storage classes to handle user commands and manage tasks.
+ */
+public class Processor {
+    private TaskList taskList;
+    private Ui chatbotUi;
+    private Storage storage;
+
+    /**
+     * Constructor for Processor element
+     * @param taskList
+     * @param chatbotUi
+     */
+    public Processor(TaskList taskList, Ui chatbotUi) {
+        this.taskList = taskList;
+        this.chatbotUi = chatbotUi;
+        this.storage = new Storage("src/data/tasks.txt", taskList);
+    }
+
+    /**
+     * Processes the given user command by delegating it to the appropriate method in the Processor class.
+     * Specifically, this method processes the user command to mark or unmark a task in the TaskList.
+     * @param userInput the user command to be processed
+     * @throws IOException if an I/O error occurs while processing the command
+     */
+    public void userInputProcessMarkUnmark(String userInput) throws IOException {
+        String[] array = userInput.split(" ");
+        String command = array[0];
+
+        
+
+        try {
+            int number = Integer.parseInt(array[1]);
+
+            if (command.equals("mark")) {
+                taskList.markTask(number - 1);
+                System.out.println(chatbotUi.dividerWrapper(
+                        chatbotUi.mark() + "\n" + taskList.getTaskAtIndex(number - 1)));
+            } else if (command.equals("unmark")) {
+                taskList.unmarkTask(number - 1);
+                System.out.println(chatbotUi.dividerWrapper(
+                        chatbotUi.unmark() + "\n" + taskList.getTaskAtIndex(number - 1)));
+            }
+            storage.writeToFile();
+        } catch (NumberFormatException e) {
+            System.out.println(chatbotUi.dividerWrapper("You must use a number to mark."));
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(chatbotUi.dividerWrapper("You must select a number within the size of the Task List."));
+        } catch (IOException e) {
+            System.out.println(chatbotUi.dividerWrapper("Error writing to file in storage."));
+        }
+    }
+
+
+    /**
```

Comment:
Noted on nice way of writing Java doc to describe on a method. Noted this on other places as well

Reply:
Appreciate it.